client/src/services/fileOpener.ts

```diff
@@ -0,0 +1,117 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+import { detectFileContentType, FileContentType } from '@/common/fileTypes';
+import { entryStat, EntryStat, entryStatAt, FsPath, getSystemPath, isDesktop, WorkspaceHandle, WorkspaceHistoryEntryStat } from '@/parsec';
+import { navigateTo, Routes } from '@/router';
+import { Information, InformationLevel, InformationManager, PresentationMode } from '@/services/informationManager';
+import { recentFileManager } from '@/services/recentFiles';
+import { DateTime } from 'luxon';
+import { Base64, openSpinnerModal } from 'megashark-lib';
+
+interface OpenPathOptions {
+  skipViewers?: boolean;
+  atTime?: DateTime;
+}
+
+const OPEN_FILE_SIZE_LIMIT = 15_000_000;
```

Comment:
Should it not be a config or env variable?
And 15 MB seems very small 🤷

Reply:
Not much use in it being an env variable since it would be defined when building anyway. We can make it a config in another PR but I don't really want a user setting 16Go and complaining that it wrecks their system.

15MB is actually quite a lot given that it's all in memory with no streaming of any kind, and that for docx or xlsx, it's converted to HTML. It would mean that in memory we have the whole docx of 15MB + whatever the docx converted to HTML weighs (I wouldn't be surprised if it's x10).